yt_dlp/extractor/common.py

```diff
@@ -220,6 +220,18 @@ class InfoExtractor:
                     * no_resume  The server does not support resuming the
                                  (HTTP or RTMP) download. Boolean.
                     * has_drm    The format has DRM and cannot be downloaded. Boolean
+                    * extra_param_to_segment_url  A query string to append to each
+                                 fragment's URL, or to update each existing query string
+                                 with. Only applied by the native HLS/DASH downloaders.
+                    * _decryption_key_url  The URI from which the AES-128 key used to
+                                 decrypt the HLS segments should be downloaded
+                                 (native HLS downloader only).
+                    * hls_aes_key  The key (as hex) to use for decrypting the HLS
+                                 segments instead of retrieving a key from the URI
+                                 given in the playlist (native HLS downloader only).
+                    * hls_aes_iv  The IV (as hex) to use for decrypting the HLS
+                                 segments instead of the IV provided in the playlist
+                                 (native HLS downloader only).
```

Comment:
maybe it makes sense to put these inside a dict? like `hls_aes: {key, url, iv}`

Then we can do this to prevent passing such formats to external downloaders
```diff
--- a/yt_dlp/downloader/external.py
+++ b/yt_dlp/downloader/external.py
@@ -104,6 +104,7 @@ def supports(cls, info_dict):
         return all((
             not info_dict.get('to_stdout') or Features.TO_STDOUT in cls.SUPPORTED_FEATURES,
             '+' not in info_dict['protocol'] or Features.MULTIPLE_FORMATS in cls.SUPPORTED_FEATURES,
+            not info_dict.get('aes_key') and not info_dict.get('extra_param_to_segment_url'),
             all(proto in cls.SUPPORTED_PROTOCOLS for proto in info_dict['protocol'].split('+')),
         ))
```

Reply:
I liked the dict idea and went with it. I tweaked the external.py code suggestion to use `traverse_obj` so that it would ignore an `hls_aes` dict with keys but no truthy values. This will eliminate the need for extractor code to *only* return an `hls_aes` dict if it knows for certain there are uri/key/iv values